packages/hardhat-core/src/internal/artifacts.ts

```diff
@@ -65,7 +65,7 @@ export class Artifacts implements IArtifacts {
     } catch (error) {
       if (
         typeof error.message === "string" &&
-        error.message.includes("no matching file exists")
+        (error.message as string).includes("no matching file exists")
```

Comment:
There are several of these, so if we end up doing what I mentioned before, we should create a helper, something like `assertErrorIsObject` or whatever.

Reply:
When doing the assertion I get the error "Property 'message' does not exist on type 'object'", can I just use any instead of unknown as the error type?

```
    } catch (error: unknown) {
      assertHardhatInvariant(
        typeof error === "object" && error !== null,
        "Error is not an object"
      );
      if (
        (error?.message?.includes("no matching file exists") as boolean) ?? //Error on error?.message
        false
      ) {
```

Reply:
When doing the assertion I get the error "Property 'message' does not exist on type 'object'".  Removing unknown declaration removes this issue (leaving it as any)

```
    } catch (error: unknown) {
      assertHardhatInvariant(
        typeof error === "object" && error !== null,
        "Error is not an object"
      );
      if (
        (error?.message?.includes?.("no matching file exists") as boolean) ?? //error here on error?.message
        false
      ) {
```